test/framework/test_composite.py

```diff
@@ -1000,3 +1021,464 @@ def circuits(self):
+            "metadata": {"test_val": 5},
+            "counts": {"0": 100, "1": 900},
+        }
+
+        self.ref_data = pd.DataFrame.from_dict(
+            {
+                "name": ["p1", "p1", "p1", "p1", "p1"],
+                "experiment": [
+                    "SomeExperiment1",
+                    "SomeExperiment1",
+                    "SomeExperiment1",
+                    "SomeExperiment1",
+                    "SomeExperiment2",
+                ],
+                "components": [[Qubit(0)], [Qubit(0)], [Qubit(1)], [Qubit(1)], [Qubit(2)]],
+                "value": [0.6, 0.5, 0.7, 0.6, 0.9],
+            }
+        )
+
+        self.ref_child_data_0_0 = pd.DataFrame.from_dict(
+            {
+                "name": ["p1", "p1"],
+                "experiment": ["SomeExperiment1", "SomeExperiment1"],
+                "components": [Qubit(0), Qubit(0)],
+                "value": [0.6, 0.5],
+            }
+        )
+
+        self.ref_child_data_0_1 = pd.DataFrame.from_dict(
+            {
+                "name": ["p1", "p1"],
+                "experiment": ["SomeExperiment1", "SomeExperiment1"],
+                "components": [Qubit(1), Qubit(1)],
+                "value": [0.7, 0.6],
+            }
+        )
+
+        self.ref_child_data_1 = pd.DataFrame.from_dict(
+            {
+                "name": ["p1"],
+                "experiment": ["SomeExperiment2"],
+                "components": [Qubit(2)],
+                "value": [0.9],
+            }
+        )
+
+        self.ref_data_not_flatten = (
+            row_iter
+            for row_iter in [
+                *tee(self.ref_child_data_0_0.iterrows(), 1),
+                *tee(self.ref_child_data_0_1.iterrows(), 1),
+                *tee(self.ref_child_data_1.iterrows(), 1),
+            ]
+        )
+
+    def test_experiment_data_bootstrap_child_flatten(self):
+
+        """
+        Checks bootstrap when flatten
+        """
+
+        exp_data = ExperimentData()
+
+        exp_data.metadata.update(
+            {
+                "component_types": ["ParallelExperiment", "SomeExperiment2"],
+                "component_metadata": [
+                    {
+                        "component_types": ["SomeExperiment1", "SomeExperiment1"],
+                        "component_metadata": [
+                            {
+                                "physical_qubits": [0],
+                                "device_components": [Qubit(0)],
+                            },
+                            {
+                                "physical_qubits": [1],
+                                "device_components": [Qubit(1)],
+                            },
+                        ],
+                    },
+                    {
+                        "physical_qubits": [2],
+                        "device_components": [Qubit(2)],
+                    },
+                ],
+            }
+        )
+
+        exp_data.add_data(self.mock_data)
```

Comment:
Are these lines common? If so you can also move this code in setup and give each test a prepared ExperimentData instance.